mitodl/open-discussions
# Added model and API to subscribe to comments and posts

#### What are the relevant tickets?
Fixes #466 
Fixes #468 
Fixes #465 

#### What's this PR do?
Adds model and API to add/remove post/comment subscriptions

#### How should this be manually tested?
Go to the APIs endpoints for a post and comment (`/api/v0/post/ID/` and `/api/v0/comment/ID` respectively) and verify you can `PATCH` those with `{"subscribed": true|false}` and the desired behavior happens and a corresponding `Subscription` record is created or deleted as appropriate.

#### Any background context you want to provide?
I implemented a `Base36IntegerField` that we can use going forward when we need to store a reddit id somewhere. It accepts base-36 strings and returns them while storing them in a bigint field the same as the reddit backend does. This make the data type opaque from an application perspective but gives us a backend representation that aligns with reddit's. This was an easier approach than manually encoding/decoding wherever was needed.
